assets/course-theme/complete-lesson-button.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * WordPress dependencies
+ */
+import domReady from '@wordpress/dom-ready';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Initializes complete lesson transition.
+ */
+export const initCompleteLessonTransition = () => {
+	domReady( () => {
+		const completeForms = document.querySelectorAll(
+			'[data-id="complete-lesson-form"], .lesson_button_form'
+		);
+		const completeButtons = document.querySelectorAll(
+			'[data-id="complete-lesson-button"], .wp-block-sensei-lms-button-complete-lesson button'
+		);
+		const progressBars = document.querySelectorAll(
+			'.sensei-course-theme-course-progress-bar-inner'
+		);
+		const mainContent = document.querySelector(
+			'.sensei-course-theme__main-content'
+		);
+
+		/**
+		 * Disable complete buttons.
+		 */
+		const disableButtons = () => {
+			completeButtons.forEach( ( button ) => {
+				button.setAttribute( 'disabled', 'disabled' );
+				button.classList.add( 'is-busy' );
+			} );
+		};
+
+		/**
+		 * Delay submit to show animations.
+		 *
+		 * @param {Object} e    The submit event.
+		 * @param {Object} form The form DOM node.
+		 */
+		const delayFormSubmit = ( e, form ) => {
+			e.preventDefault();
+			disableButtons();
+
+			setTimeout( () => {
+				form.submit();
+			}, 1000 );
+		};
+
+		/**
+		 * Run progress bar animation.
+		 */
+		const runProgressBarAnimation = () => {
+			progressBars.forEach( ( progressBar ) => {
+				const { completed, count } = progressBar.dataset;
+
+				// Percentage with one more completed.
+				const percentage = ( ( +completed + 1 ) / +count ) * 100;
```

Comment:
what do these + prefixes in `completed` and `count` do? O.o
Genuinely curious...I think I have to up my JS game haha